integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -330,7 +334,14 @@ private int readdirInternal(String path, long buff, long filter, long offset,
       FuseFillDir.apply(filter, buff, "..", null, 0);
 
       mFileSystem.iterateStatus(uri, file -> {
-        FuseFillDir.apply(filter, buff, file.getName(), null, 0);
+        ByteBuffer buffer = ByteBuffer.allocateDirect(mFileStatSize);
```

Comment:
Can you try to reuse this buffer, so you only create one per listing please? We can assume to apply function does not keep a copy of it, otherwise there would be a memory leak.